engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -195,17 +197,21 @@ Object doPolyglot(
     boolean anyWarnings = false;
     ArrayRope<Warning> accumulatedWarnings = new ArrayRope<>();
     for (int i = 0; i < argExecutors.length; i++) {
-       var r = argExecutors[i].executeThunk(arguments[i + 1], state, TailStatus.NOT_TAIL);
-       args[i] = r;
+      var r = argExecutors[i].executeThunk(arguments[i + 1], state, TailStatus.NOT_TAIL);
       if (r instanceof DataflowError) {
         profiles[i].enter();
         return r;
-      } else if (r instanceof WithWarnings w) {
+      } else if (warnings.hasWarnings(r)) {
```

Comment:
This is certainly better coding style. We are close to fixing multi-dimensional arrays, I think. Can you test invocation of a Java method that accepts `long[][]` and there is a `Number` with warning somewhere deep in the 2nd level Enso array?

Reply:
Yeah, the current approach doesn't work with multi-dimensional arrays, yet. I will see if I can do anything about it.

Reply:
Now we do.